mesonbuild/compilers/cpp.py

```diff
@@ -67,6 +68,49 @@ def has_header_symbol(self, hname, symbol, prefix, env, extra_args=None, depende
         int main () {{ return 0; }}'''
         return self.compiles(t.format(**fargs), env, extra_args, dependencies)
 
+    def _test_cpp_std_arg(self, cpp_std_value):
+        # Test whether the compiler understands a -std=XY argument
+        assert(cpp_std_value.startswith('-std='))
+
+        CPP_TEST = 'int main() { return static_cast<int>(0); }'
+        with self.compile(code=CPP_TEST, extra_args=[cpp_std_value], mode='compile') as p:
+            if p.returncode == 0:
+                mlog.debug('Compiler accepts {}:'.format(cpp_std_value), 'YES')
+                return True
+            else:
+                mlog.debug('Compiler accepts {}:'.format(cpp_std_value), 'NO')
+                return False
```

Comment:
Why doesn't this use `cc.has_argument()`?

Reply:
Left a comment why `has_multi_arguments()` is not an option, and even if it were, why we would want to avoid it.